### PR TITLE
Fix #227. Make button utility classes override more specific rules (e.g. in the Navbar menu)

### LINKED ISSUE #227: navbar-btn margin not vertically even
Looks like the total height for the navbar-btn is shorter than the navbar itself causing it not be in the middle of the bar: 
screen shot 2016-08-22 at 2 09 08 pm

Changing the margin to 11px from 8px might do the trick.

### PR DESCRIPTION
**Pull request by nathan-rollins**
These utility classes are being overridden by more specific rules for anchors in the Navbar menu, particularly when used for a Navbar Button.

While I'm not keen on using !important, this keeps the rules more tidy, and perhaps defensible for a utility class. The alternative, apart from refactoring the navigation menu rules, is to include specific selectors (.navbar-ws .navbar-nav > li > a...) for each of these classes to permit Navbar buttons to format as intended.

### CODE REVIEW

**Comment by nathan-rollins** [comment]
And maybe to be particularly safe, it might be good to return to wraith and see if we can get it cooperate.

**Comment by rraub** [comment]
`!important` ... 
giphy 22

**Comment by rraub** [comment]
So I'm trying to do some research on the bootstrap lint warning [E043](https://github.com/twbs/bootlint/wiki/E043) for using an anchor tag with `btn` within the nav. I haven't seen any problems with it, and I haven't uncovered the source for that bootlint error.

**Comment by rraub** [comment]
Also I thought we could only use the gold buttons in the nav, but according to the style guide its [discouraged, but not disallowed](https://brandguide.asu.edu/web-standards/enterprise/navigation)

 [screen shot 2016-10-04 at 9 40 30 am](https://cloud.githubusercontent.com/assets/295804/19083393/bb0cbf36-8a16-11e6-831a-d7a3d4955c80.png)

**Comment by rraub** [comment]
Also we should change the hover, active, and focus colors to go lighter rather than darker since these buttons are on the dark bar they kind of fade into the bar. 
FYI: [ASUOnline](https://asuonline.asu.edu/) uses `#ffc13b` for hover on the gold button, that doesn't look bad.

**Comment by rraub** [comment]
Here are some docs I wrote that we should add in (and maybe modify further):
```
// Navbar Buttons
//
// Some Buttons can be used in the navbar.
// NOTE: This goes against <a href="https://github.com/twbs/bootlint/wiki/E043">bootstraps
// recommendation</a> however in v4 these restrictions have been lifted. 
//
// Markup:
// <nav class="navbar navbar-ws affix-top" role="navigation">
//     <div class="container-fluid">
//         <div class="collapse navbar-collapse">
//             <ul class="nav navbar-nav">
//                 <li><a href="#">Normal</a></li>
//                 <li><a href="#" class="btn btn-gold">Gold</button></a></li>
//                 <li><a href="#" class='btn btn-primary'>Primary</a></li>
//                 <li><a href="#">Normal</a></li>
//             </ul>
//         </div>
//     </div>
// </nav>
//
// Styleguide navigation.web-standards.buttons
```

**Comment by rraub** [comment]
final thought: We should add in some styles for the buttons in the mobile nav. Currently they look a little miss placed in the mobile menu, it might be best to have them go fill width with a margin.

**Comment by nathan-rollins** [comment]
Oh, forgot to fix the mobile styling. That is incoming.

**Comment by nathan-rollins** [comment]
Preview of the mobile menu:

screen shot 2016-10-11 at 4 52 32 pm

**Comment by nathan-rollins** [comment]
Here's an updated preview without the rounded corners in the dropdown menu.

screen shot 2016-10-11 at 5 31 47 pm

**Comment by nathan-rollins** [comment]
There, I think that looks pretty nice. I can load this up on my MAMP instance if you would like to try it on your phone.

File: scss/navigation/_ws-navbar.scss [hunk 1]
```diff
@@ -110,6 +112,10 @@
         }
       }
 
+      li.menu-btn-navbar {
```
Comment (rraub) [inline scss/navigation/_ws-navbar.scss]: are we using this style?

Comment (nathan-rollins) [inline scss/navigation/_ws-navbar.scss]: Yes, it is used on the enclosing <li> for the button anchor.

**Comment by rraub** [comment]
Looks Awesome! Well done!

**Pull request merged**